### PR TITLE
automatically set pool.max and pool.min to integer

### PR DESCRIPTION
**Pull request by knoxcard**
automatically set pool.max and pool.min to integer, if they are defined and `> -1`, or else set safe defaults instead of throwing errors

### CODE REVIEW

**Review by knoxcard — COMMENTED** [review]
missing semicolons? why are these Travis tests failing?

**Review by knoxcard — COMMENTED** [review]
F U Travis, lol

**Review by knoxcard — COMMENTED** [review]
Travis, I am going to whoop yo ass boy!

**Review by knoxcard — COMMENTED** [review]
would love to get some sleep...

**Review by knoxcard — COMMENTED** [review]
blah

**Review by knoxcard — COMMENTED** [review]
bleh

**Review by knoxcard — COMMENTED** [review]
surprise me

**Review by knoxcard — COMMENTED** [review]
Number()

**Comment by knoxcard** [comment]
**OMG FINALLY, i could have rewritten the entire NodeJS framework in that amount of time.**

Greenkeeper = good bot
Travis = crap bot

Turned 14 lines of code into 2 lines...good riddance

**Comment by DiegoRBaquero** [comment]
You removed validations (e.g. what if it's less than 0) and added forced casting to number. But the numbers need to be checked to be > 0 and Int.

**Comment by knoxcard** [comment]
@DiegoRBaquero - all checks passed, is that particular logic you mentioned not included in the test run?

**Comment by DiegoRBaquero** [comment]
I guess not, tests should be added for such scenarios and check that it fails.

**Comment by knoxcard** [comment]
gotchya, i will alter the code now...

**Review by knoxcard — COMMENTED** [review]
Set safe defaults when min, max conditions are not met
min =1
max = 5

Love keeping it to just two lines of code, but this might not suffice.

Thoughts? @sushantdhiman

**Comment by knoxcard** [comment]
@DiegoRBaquero
`> 0` failed, appears that zero is inclusive, a valid option.  `> -1` passed the tests.

**Review by knoxcard — COMMENTED** [review]
nevermind, throw new error, don't set safe defaults

**Review by knoxcard — COMMENTED** [review]
lord give me patience...

**Review by knoxcard — COMMENTED** [review]
petty prettier requirements

**Review by knoxcard — COMMENTED** [review]
semicolons suck

**Review by knoxcard — COMMENTED** [review]
poop

**Comment by tiblu** [comment]
@DiegoRBaquero @knoxcard Need any help pushing this forward? I think it's a good idea, because everything set from ENV is a string by default, is see no harm in `Number`/`parseInt` and then check the other conditions.
Linking our related issue here - https://github.com/citizenos/citizenos-api/issues/137

File: lib/Pool.js [hunk 1]
```diff
@@ -54,22 +54,17 @@ class Pool {
...
-      factory.max <= 0 ||
-      factory.max !== Math.round(factory.max)
-    ) {
-      throw new Error("max must be an integer > 0");
+    
+    if (!factory.max) {
+      throw new Error("factory.max must be >= 0");
     }
-
+
```
Comment (DiegoRBaquero) [inline lib/Pool.js]:
```suggestion

```

**Comment by tiblu** [comment]
@DiegoRBaquero Any plans on implementing this? I understand that probably there are many other things to do, but I need to know what's going to happen with this. If you have no plans of implementing this, then we'll just move on with other solutions.
All the best!

**Comment by papb** [comment]
Hello everyone, what is the use case for having sequelize-pool convert these configuration values to Number? I did not understand. Why can't the user enforce the correct types? Why would an user pass something that is not a number here?

**Pull request closed without merging**